src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -326,14 +326,15 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     const SchedulingKey &scheduling_key, const rpc::Address *raylet_address) {
   auto &scheduling_key_entry = scheduling_key_entries_[scheduling_key];
 
-  if (scheduling_key_entry.pending_lease_requests.size() ==
-      max_pending_lease_requests_per_scheduling_category_) {
+  uint64_t max_pending_lease_requests =
+      std::max(get_max_pending_lease_requests_per_scheduling_category_(),
+               static_cast<uint64_t>(1));
```

Comment:
why do we have this `max(get_max_..., 1)`? Seems that `get_max_...` will always return 10? seems that we should assert false when this returns 0.